midterm-2023-11-06/code/5115-nikola-tosic/script.ts

```diff
@@ -0,0 +1,162 @@
+    const filter = document.getElementById("country-filter")!;
+
+    const countries = new Set<string>();
+    for (const band of bands) {
+        countries.add(band.country);
+    }
+
+    for (const country of countries) {
+        const option = document.createElement("option");
+        option.value = country;
+        option.innerHTML = country;
+        filter.appendChild(option);
+    }
+};
+
+const applyFilter = () => {
+    const countryElement = document.getElementById("country-filter") as HTMLSelectElement;
+    const country = countryElement.value;
+
+    let filteredBands = bands;
+    if (country !== "all") {
+        filteredBands = filteredBands.filter(band => band.country === country);
+    }
+
+    displayBands(filteredBands);
+};
+
+const loadData = async () => {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json";
+    const response = await fetch(dataUri);
+
+    if (!response.ok) {
+        throw new Error("Failed to fetch data");
+    }
+
+    const data = await response.json();
+    return data.metalBands;
+};
+
+const displayBands = (bands: Band[]) => {
+    const container = document.getElementById("band-container")!;
+    container.innerHTML = "";
+    for (const band of bands) {
+        const bandRow = generateBandRow(band);
+        container.appendChild(bandRow);
+    }
+};
+
+const generateBandRow = (band: Band) => {
+    const row = document.createElement("div");
+    row.classList.add("band-row");
+
+    const idCell = document.createElement("div");
+    idCell.classList.add("band-data", "band-id");
+    idCell.innerHTML = band.id.toString();
+    row.appendChild(idCell);
+
+    const nameCell = document.createElement("div");
+    nameCell.classList.add("band-data", "band-name");
+    nameCell.innerHTML = band.name;
+    row.appendChild(nameCell);
+
+    const formedCell = document.createElement("div");
+    formedCell.classList.add("band-data", "band-formed");
+    formedCell.innerHTML = band.formed.toString();
+    row.appendChild(formedCell);
+
+    const locationCell = document.createElement("div");
+    locationCell.classList.add("band-data", "band-location");
+    locationCell.innerHTML = band.location;
+    row.appendChild(locationCell);
+
+    const countryCell = document.createElement("div");
+    countryCell.classList.add("band-data", "band-country");
+    countryCell.innerHTML = band.country;
+    row.appendChild(countryCell);
+
+    const membersCell = document.createElement("div");
+    membersCell.classList.add("band-data", "band-members");
+    membersCell.innerHTML = band.members.join(", ");
+    row.appendChild(membersCell);
+
+    const albumsCell = document.createElement("div");
+    albumsCell.classList.add("band-data", "band-albums");
+    albumsCell.innerHTML = band.albums.length.toString();
+    albumsCell.addEventListener("click", () => {
+        const modal = document.getElementById("album-details")!;
```

Comment:
The albums being displayed in a modal is quite ok, but the issue is that the modal does not exist in the html. Only imagining it, this is +10 points.